sw/device/tests/hmac_enc_irq_test.c

```diff
@@ -0,0 +1,147 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_hmac.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/hmac_testutils.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static const uint32_t kPlicTarget = kTopEarlgreyPlicTargetIbex0;
+
+dif_hmac_t hmac;
+dif_rv_plic_t plic;
+
+/**
+ * Flags to check whether the interrupts have fired.
+ */
+static bool hmac_empty_irq;
+static bool hmac_done_irq;
+
+const test_config_t kTestConfig;
+
+HMAC_TESTUTILS_ASSERT_LITTLE_ENDIAN;
+static const dif_hmac_transaction_t kHmacTransactionConfig = {
+    .digest_endianness = kDifHmacEndiannessLittle,
+    .message_endianness = kDifHmacEndiannessLittle,
+};
+
+/**
+ * Services HMAC interrupts.
+ *
+ * For the HMAC done the interrupt is cleared internally by `dif_hmac_finish`.
+ */
+static void handle_hmac_isr(const dif_rv_plic_irq_id_t interrupt_id) {
+  // NOTE: This initialization is superfluous, since the `default` case below
+  // is effectively noreturn, but the compiler is unable to prove this.
+  dif_hmac_irq_t hmac_irq = 0;
+
+  switch (interrupt_id) {
+    // Digest verification is done in a separate test, thus the digest is
+    // discarded.
+    case kTopEarlgreyPlicIrqIdHmacHmacDone:
+      hmac_irq = kDifHmacIrqHmacDone;
+      dif_hmac_digest_t dummy_digest;
+      dif_result_t res = dif_hmac_finish(&hmac, &dummy_digest);
+      CHECK(res == kDifOk, "HMAC finish failure = %d", res);
```

Comment:
Isn't there a CHECK_DIF_OK you can use?

Reply:
This change is part of the previous PR (lowRISC#10960)

Reply:
Done in the other PR.